### PR TITLE
Add web3 providers: ConnectKit and Wagmi, add working wallet

### PR DESCRIPTION
**Pull request by daedboi**
Added 3 new dependencies: wagmi, connectkit and ethers.

Implemented web3 client in _app.tsx with custom FTM chain.

Replaced wallet button with ConnectKit button in the Header.

Since you had added logic for detecting if the wallet's connected (ethereum injected), probably better that we replace this with Wagmi's solution: https://wagmi.sh/docs/hooks/useAccount

Left old replaced code commented out for now

### CODE REVIEW

File: src/components/Header/index.tsx [hunk 1]
```diff
@@ -47,7 +48,11 @@ const Header = () => {
               <img src="/images/icons/icon-logo.svg" alt="" />
             </div>
             <p>$0.00</p>
-            <Link href="#">
+
+            {/* Using ConnectKit wallet button */}
+            <ConnectKitButton />
```
Comment (nikmace) [inline src/components/Header/index.tsx:53]: There is a HTML for the button. So the expected behavior should be just a button when we are not connected. When we connect a wallet it should dynamically change to a block that contains the coin amount plus the connected wallet address.

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -7,38 +7,80 @@ import superjson from 'superjson';
...
 import '../scss/main.scss';
 import { URL } from '../constants';
 import { AppRouter } from '../server/route/app.router';
 import { store } from '../redux/store';
 import Layout from '../components/Layout';
 
+const ftmScan = { name: 'Ftmscan', url: 'https://ftmscan.com/' }; // blockexplorer for Fantom
+
+// Since Fantom is not a default chain supported by wagmi, we need to import the Chain type and create our own
+const ftm: Chain = {
```
Comment (nikmace) [inline src/pages/_app.tsx:22]: These const things should be placed in a constants file or utils/wallet.ts

**Review by nikmace — CHANGES_REQUESTED** [review]
Errors in the console in dev tools should be examined.

UI looks fine, nothing breaks. But as said there is an HTML where the button should be placed (or a div). Plus the logic when you connect a wallet should be implemented.

File: src/components/Header/index.tsx (hunk 1 above)
Comment (daedboi) [reply src/components/Header/index.tsx:51]: Will be leaving token price to show at all times, as not influenced by wallet status.

**Review by nikmace — APPROVED** [review]

**Pull request merged**